db/write_thread.cc

```diff
@@ -214,11 +217,35 @@ void WriteThread::SetState(Writer* w, uint8_t new_state) {
   }
 }
 
-bool WriteThread::LinkOne(Writer* w, std::atomic<Writer*>* newest_writer) {
+bool WriteThread::LinkOne(
+    Writer* w,
+    std::atomic<Writer*>* newest_writer,
+    const bool check_write_stall) {
```

Comment:
how about always check for the dummy and remove this extra param? I think that way it make the code look a little bit simpler. The only callsite where you set this param to false is for the memtable writer queue, and the dummy won't be appearing in that case anyway.

Reply:
I missed the call site at `BeginWriteStall` but that one won't expect to see a dummy either.